### PR TITLE
More on compromises; important for academics

### PR DESCRIPTION
**Pull request by standage**

### CODE REVIEW

File: short/index.html [hunk 1]
```diff
@@ -357,27 +374,16 @@
 
 *   Write **lots** of tests for scientific core of the code
 *   Start with many fewer tests for other project components
-    *   command-line interface
-    *   file I/O
-    *   other "boring" and "unimportant" code
 *   When bugs are encountered:
+    *   write new regression tests specific to that bug  <!-- #tdd #ftw -->
     *   fix the observed bug
-    *   write new regression tests specific to that bug
```
Comment (standage) [inline short/index.html:365]: In my LBL talk, somebody reminded me that this was a great place to put TDD into practice. Duh!

Comment (standage) [reply short/index.html:365]: This comment only makes sense in the larger context. I wish GitHub would let me select which lines to include in the comment...

File: short/index.html [hunk 2]
```diff
@@ -347,7 +347,24 @@
...
         *   optimize code surgically
     *   Clean up code
-*   Compromise on best practices if you must
+</section>
+
+<section data-markdown>
+**Compromises (cont.)**
+
+*   Incremental improvement ([Petre](http://arxiv.org/abs/1407.5648))
+    *   Consider NOT trying to fix everything at once
```
Comment (gvwilson) [inline short/index.html]: I would strengthen this: "Don't try to fix everything at once"

File: short/index.html [hunk 3]
```diff
@@ -357,27 +374,16 @@
 
 *   Write **lots** of tests for scientific core of the code
 *   Start with many fewer tests for other project components
-    *   command-line interface
-    *   file I/O
-    *   other "boring" and "unimportant" code
```
Comment (gvwilson) [inline short/index.html:362]: Why remove these examples?

File: short/index.html [hunk 4]
```diff
@@ -357,27 +374,16 @@
...
-*   Avoids wasting time writing tests for bugs that:
-    *   never appear, or appear rarely
-    *   don't affect correctness of results
-*   Order of priorities
-    *   correctness
-    *   performance
-    *   user experience
-    *   ...
-    *   beautiful code
+Avoids wasting time writing tests for bugs that never appear!
```
Comment (gvwilson) [inline short/index.html:386]: Why cut the list? I think both points and their sub-points are important.

Comment (standage) [reply short/index.html:386]: I agree they're important points, but 90-120 minutes limits the number of important points you can make, and the amount of text the audience can handle. In the spirit of the material, I had to make some compromises. :-)

File: short/index.html (hunk 2 above)
Comment (standage) [reply short/index.html]: Done.

File: short/index.html (hunk 4 above)
Comment (standage) [reply short/index.html:386]: Maybe these points should be left in the master slides, with the understanding that the material may have to be trimmed for a particular audience or time constraint.

**Pull request merged**